docs/integrations/prometheus.txt

```diff
@@ -0,0 +1,175 @@
++--------------------+
+| Determined 0.17.8+ |
++--------------------+
+
+This document describes the setup and configuration to enable a
+Grafana dashboard to monitor Determined hardware and system metrics on a cloud
+cluster, such as AWS or Kubernetes. Determined
+provides a Prometheus endpoint that contains mappings between internal
+task, GPU, and container definitions, which are used by Prometheus to
+collect relevant metrics on a cluster running Determined. The endpoint
+is not enabled by default but can be enabled in the master configuration
+file.
+
+***********
+ Reference
+***********
+
+`Grafana <https://grafana.com/docs/grafana/latest/installation/>`__
+
+`Prometheus
+<https://prometheus.io/docs/prometheus/latest/installation/>`__
+
+`cAdvisor
+<https://github.com/google/cadvisor/blob/master/docs/storage/prometheus.md>`__
+
+`dcgm-exporter <https://github.com/NVIDIA/dcgm-exporter>`__
+
+*************
+ Constraints
+*************
+
+The Determined Prometheus endpoint is configured to work with `cAdvisor
+<https://github.com/google/cadvisor>`__ for CPU metrics and
+`dcgm-exporter <https://github.com/NVIDIA/dcgm-exporter>`__ (DCGM) for
+GPU metrics. Although other monitoring tools can be used with the setup,
+this guide describes only cAdvisor and DCGM tool configuration.
+Prometheus queries on metrics collected by other tools can differ,
+depending on the format and organization of the returned metrics.
+
+***************
+ Prerequisites
+***************
+
+-  A Grafana installation for dashboard monitoring.
+-  An on-cluster Prometheus instance for time-series data collection.
+
+**********************
+ Configure Determined
+**********************
+
+Install and run Determined on a cluster. When launching the master
+instance, enable the Prometheus endpoints by adding a flag to the
+``master.yaml`` configuration file:
+
+.. code:: yaml
+
+   observability:
+       enable_prometheus: true
+
+This enables the following two Prometheus API endpoints on the instance.
+
+-  ``{$DET_MASTER_ADDR}/prom/det-state-metrics``:
+
+   The ``det-state-metrics`` endpoint includes various machine-level
+   label mappings to internal Determined entities, such as GPU UUIDs and
+   container IDs to task, allocation, and experiment labels, which are
+   used by PromQL to join vectors.
+
+-  ``{$DET_MASTER_ADDR}/prom/det-http-sd-config``:
+
+   The ``det-http-sd-config`` endpoint contains address and resource
+   pool information for currently active agents. These are used by
+   Prometheus as targets for scraping. This endpoint is configured to
+   support running default cAdvisor, port 8080, and DCGM, port 9400,
+   monitoring. Other tools exposing Prometheus metrics can be used
+   instead of cAdvisor and DCGM if they are running on these ports.
+
+**************************************
+ Configure cAdvisor and dcgm-exporter
+**************************************
+
+The cAdvisor and dcgm-exporter tools must be running on the cluster
```

Comment:
maybe we should add "monitoring" after "cAdvisor and dcgm-exporter"/before "tools" here? feels like "tools" should have another adjective

Reply:
done